sharding-orchestration/sharding-orchestration-core-new/src/test/java/org/apache/shardingsphere/orchestration/temp/internal/keygen/LeafSnowflakeKeyGeneratorTest.java

```diff
@@ -0,0 +1,272 @@
+        Properties properties = new Properties();
+        properties.setProperty("serverList", "127.0.0.1:2181");
+        properties.setProperty("serviceId", "testService1");
+        properties.setProperty("maxTimeDifference", "5000");
+        properties.setProperty("distributedLockManagementType", "ForthTestRegistryCenter");
+        leafSnowflakeKeyGenerator.setProperties(properties);
+        FieldUtil.setStaticFinalField(leafSnowflakeKeyGenerator, "timeService", new FixedTimeService(1));
+        List<Comparable<?>> expected = Collections.<Comparable<?>>singletonList(4198401L);
+        List<Comparable<?>> actual = new ArrayList<>();
+        actual.add(leafSnowflakeKeyGenerator.generateKey());
+        assertThat(actual, is(expected));
+    }
+    
+    @Test(expected = IllegalStateException.class)
+    public void generateKeyFailureWithTimeRollback() {
+        Properties properties = new Properties();
+        properties.setProperty("serverList", "127.0.0.1:2181");
+        properties.setProperty("serviceId", "specialService");
+        properties.setProperty("maxTimeDifference", "5000");
+        properties.setProperty("distributedLockManagementType", "FifthTestRegistryCenter");
+        leafSnowflakeKeyGenerator.setProperties(properties);
+        FieldUtil.setStaticFinalField(leafSnowflakeKeyGenerator, "timeService", new PreviousTimeService(15000));
+        leafSnowflakeKeyGenerator.generateKey();
+    }
+    
+    @Test
+    public void generateKeySuccessWithTimeRollback() {
+        Properties properties = new Properties();
+        properties.setProperty("serverList", "127.0.0.1:2181");
+        properties.setProperty("serviceId", "specialService");
+        properties.setProperty("maxTimeDifference", "5000");
+        properties.setProperty("distributedLockManagementType", "FifthTestRegistryCenter");
+        leafSnowflakeKeyGenerator.setProperties(properties);
+        FieldUtil.setStaticFinalField(leafSnowflakeKeyGenerator, "timeService", new PreviousTimeService(3000));
+        leafSnowflakeKeyGenerator.generateKey();
+    }
+    
+    @Test
+    public void assertGenerateKeyWithMultipleThreads() throws Exception {
+        int threadNumber = Runtime.getRuntime().availableProcessors() << 1;
+        ExecutorService executor = Executors.newFixedThreadPool(threadNumber);
+        Properties properties = new Properties();
+        properties.setProperty("serverList", "127.0.0.1:2181");
+        properties.setProperty("serviceId", "testService1");
+        properties.setProperty("maxTimeDifference", "5000");
+        properties.setProperty("distributedLockManagementType", "ForthTestRegistryCenter");
+        leafSnowflakeKeyGenerator.setProperties(properties);
+        Set<Comparable<?>> actual = new HashSet<>();
+        int taskNumber = threadNumber << 2;
+        for (int i = 0; i < taskNumber; i++) {
+            actual.add(executor.submit(new Callable<Comparable<?>>() {
+
```

Comment:
I think this blank line should remove.

Reply:
THX. I have resolved it.

Reply:
I‘m sorry, I make a mistake. This blank line should be remained and don't need remove.